python_modules/dagster/dagster/_utils/warnings.py

```diff
@@ -35,6 +35,31 @@ def preview_warning(
     )
 
 
+# ########################
+# ##### BETA
+# ########################
+
+
+class BetaWarning(Warning):
+    pass
+
+
+def beta_warning(
+    subject: str,
+    additional_warn_text: Optional[str] = None,
+    stacklevel: int = 3,
+):
+    if not _warnings_on.get():
+        return
+
+    warnings.warn(
+        f"{subject} is in beta stage - the feature is still being tested and may change."
```

Comment:
this seems ok - take it or leave it
> is in beta stage - the feature is being tested for production readiness and may be updated.

Reply:
How about:

> is in beta. It may break in a future minor release, or have minor behavior changes between dot releases.

I don't love this "production readiness" framing, as I think the key important detail here is when / how likely this feature is to be broken, and what those breakages might look like

In my mind:

- preview: might break even between dot releases
- beta: might break on a dot release, or have behavior changes